libpyhat/regression/regression.py

```
@@ -73,66 +70,24 @@ def __init__(self, method, yrange, params, i=0):  #TODO: yrange doesn't currentl
-            self.reduce_dim = params[i]['reduce_dim']
-            self.n_components = params[i]['n_components']
-            # create a temporary set of parameters
-            params_temp = copy.copy(params[i])
-            # Remove parameters not accepted by Gaussian Process
-            params_temp.pop('reduce_dim')
-            params_temp.pop('n_components')
-            self.model = GaussianProcessRegressor(**params_temp)
-
-        if self.method[i] == 'GBR':
-            self.model = GradientBoostingRegressor(**params[i])
 
     def fit(self, x, y, i=0):
-        # if gaussian processes are being used, data dimensionality needs to be reduced before fitting
-        if self.method[i] == 'GP':
-            if self.reduce_dim == 'FastICA':
-                print('Reducing dimensionality with ICA')
-                do_ica = FastICA(n_components=self.n_components)
-                self.do_reduce_dim = do_ica.fit(x)
-            if self.reduce_dim == 'PCA':
-                print('Reducing dimensionality with PCA')
-                do_pca = PCA(n_components=self.n_components)
-                self.do_reduce_dim = do_pca.fit(x)
-
-            x = self.do_reduce_dim.transform(x)
-        #try:
-            print('Training model...')
         try:
             self.model.fit(x, y)
             self.goodfit = True
-            #print(self.model)
         except:
             self.goodfit = False
-            if self.method[i] == 'GP':
-                print('Model failed to train! (For GP this does not always indicate a problem, especially for low numbers of components.)')
-                pass
-            else:
-                print('Model failed to train!')
-                traceback.print_stack()
-
-        if self.ransac:
-            self.outliers = np.logical_not(self.model.inlier_mask_)
-            print(str(np.sum(self.outliers)) + ' outliers removed with RANSAC')
-
+            print('Model failed to train!')
```

As above, I would raise an exception here using some type and then go from there.